### ISSUE TITLE
Could you add Istio ambient to your report?

### ISSUE THREAD

**Issue by linsun** [livewyer-ops/poc-servicemesh2024#2]
### Describe the bug

Love your [report](https://github.com/livewyer-ops/poc-servicemesh2024/blob/main/docs/test-report.md), but a req - can you also add ambient there? see: https://istio.io/latest/blog/2024/ambient-reaches-beta/

### Evidence

NA

### Environment

NA

**Comment by oleksandr-polishchuk1** [comment]
Hi @linsun! Thanks for your engagement, and I'm pleased that you liked the Test Report.
We are happy to take onboard your request. I've discussed it with the team and we will run a part two of this service mesh comparison, where we will compare Istio Sidecar to Istio Ambient. I'm excited to see how Istio Ambient performs.

**Comment by linsun** [comment]
Thank you @oleksandr-livewyer! It is very cool to have a neutral party running performance and do it in the open. We appreciate it. We have other feedbacks - will continue to send it via github issues?

**Comment by oleksandr-polishchuk1** [comment]
@linsun Thanks for your appreciation. We are happy to contribute to this and have already started working on it. We are planning to run some preparatory actions today and hopefully run tests next week. In terms of feedback, feel free to provide other bits using GitHub Issues.

**Comment by linsun** [comment]
Hi @oleksandr-livewyer Thanks so much!  Feel free to let me know if you have any questions related to Istio or ambient.

Regarding other feedback, please correct me if I am wrong here as I'm learning in this space as well:

- When looking at your performance tests, I think you are testing HTTP for Linkerd and Istio, but for Cilium based on your [template](https://github.com/livewyer-ops/poc-servicemesh2024/blob/main/templates/cilium-templates.yaml), you are not doing any L7 processing (other than at the ingress gateway).  Basically, the requests among the bookinfo sample app is HTTP for Linkerd and Istio, while it is TCP for Cilium. Is it possible to test HTTP for all meshes, and/or TCP for all meshes for consistency?

- FIPS compliant. My understanding is Cilium doesn't currently support FIPS, because wireguard isn't FIPS compliant. If you google "wireguard FIPS compliant", you'd find a lot of search returns such as [https://www.reddit.com/r/WireGuard/comments/yx4ipy/is_there_an_intention_to_get_wireguard_fips/](https://www.reddit.com/r/WireGuard/comments/yx4ipy/is_there_an_intention_to_get_wireguard_fips/).

- mTLS, only Linkerd and Istiod support mTLS. Cilium, not yet, see [https://github.com/cilium/cilium/issues/25216#issuecomment-1607159389](https://github.com/cilium/cilium/issues/25216#issuecomment-1607159389).

- Cilium CPU measurement doesn't seem to account for the datapath being in kernel? AFAIK, that is not attributed to any of the Cilium daemonsets.

Many thanks for your great work!

**Comment by oleksandr-polishchuk1** [comment]
Hi @linsun ! Thank you for bringing up many valid points.
Our main goal was to provide a comparison between the three products, deploying them according to their product's official documentation while balancing that with environment and deployment consistency wherever possible. An additional benefit of this comparison was to further our engineering team's understanding of these three products, to better provide knowledge and guidance to our customers.
We will likely run another round of tests later this year, and we will take into account your and others' feedback to provide better insights.

Diving into details I can say:
* Thanks for noting this. We followed the Cilium documentation, so we will review our setup and correct it if needed before the next round of tests.
* I agree. It seems that Wireguard is not FIPS compliant. However, I bring you back to the point of our study. We did not intend to compare FIPS-compliant versions of products; we used the basic builds from the official documentation. Regarding Cilium's FIPS compliance, they provide multiple statements that they are FIPS compliant.
             [Cilium](https://cilium.io/industries/financial-services/#:~:text=Cilium%20empowers%20teams%20with%20tools%20to%20ensure%20continuous%20compliance%20with%20standards%20like%20FIPS%2C%20PCI%20DSS%2C%20Open%20Banking%2C%20and%20SOC%202%20by%20offering%20real%2Dtime%20monitoring%20and%20analysis%20of%20workload%20communication%20to%20identify%20non%2Dcompliant%20connections.)
             [Cilium](https://cilium.io/industries/security/#:~:text=Leveraging%20eBPF%2C%20Cilium%20offers%20efficient%20observability%20across%20the%20entire%20application%20stack%2C%20integrates%20seamlessly%20with%20SIEM%20systems%2C%20and%20ensures%20compliance%20with%20standards%20like%20FIPS%2C%20FedRAMP%2C%20and%20SOC)
             [Isovalent](https://isovalent.com/features/transparent-encryption/#:~:text=Meet%20compliance%20requirements%3A%20standards%20such%20as%20PCI%20and%20FIPS%20require%20encryption%20of%20data%20in%20transit.%C2%A0)
* I agree. Cilium has its own implementation of mTLS, called mTLS-based mutual authentication. I believe it solves the same issues as common mTLS. To mitigate this, we will add some notes to the test report to make this clearer to readers.
* That's a fair point. We didn't include this commentary in our test report, so I'll make sure to add this.
